akka-projection-r2dbc/src/main/scala/akka/projection/r2dbc/internal/R2dbcProjectionImpl.scala

```diff
@@ -362,9 +363,16 @@ private[projection] object R2dbcProjectionImpl {
                 Some(loadedEnvelope)
               }
             } else {
+              triggerReplayIfPossible(sourceProvider, env)
               Future.successful(None)
             }
           }
+          // FIXME not sure how to test this or if it is the right thing
+          .recover {
+            case ex: IllegalStateException if ex.getMessage.contains("Rejected envelope from backtracking") =>
```

Comment:
should be fine with all IllegalStateException without looking at the exception message. we could create a more specific exception as well, but I think both cases of IllegalStateException should trigger replay

Reply:
Done in [df490be](https://github.com/akka/akka-projection/pull/819/commits/df490be8130c9eb68f1d1e81b2fba47e8aa5af79)

One thing I'm unsure about is if we need the replays in more of the adapted handler factories in `R2dbcProjectionImpl` or not. This is currently only for flow.